backend/reloadgen.ml

```diff
@@ -63,7 +63,7 @@ method reload_operation op arg res =
      res to be stack-allocated, but do something for
      stack-to-stack moves *)
   match op with
-    | Imove | Ireload | Ispill | Iintofvalue | Ivalueofint | Ivectorcast _ ->
```

Comment:
The reinterpret casts between ints and float32/float will never have the same stack class, so this is equivalent

Reply:
I should probably appear as a comment in
the code, perhaps specifying what "int"
means: it does not soundstrictly impossible
to have the same class for unboxed int32
and unbox float.

Reply:
Sorry, that wasn't quite right. I mean currently floats / ints have a different stack class, so this will never apply, but even if we combine them this condition should be applicable - there's no transformation that must be done to the stack slot.
But either way, the amd64 reload intercepts reinterpret casts other than value<->int anyway, since register behavior matters for gpr<->xmm moves.

Added a comment saying this